addon/utils/decorator-wrappers.js

```diff
@@ -24,8 +24,17 @@ export function decoratorWithParams(fn) {
     if (isDescriptor(params[params.length - 1])) {
       return handleDescriptor(...arguments, fn);
     } else {
-      return function(/* target, key, desc */) {
-        return handleDescriptor(...arguments, fn, params);
+      return function(target, key, desc) {
+        const hasDecorator = (typeof params[0]) === "function";
+        const decorators = hasDecorator ? params.filter(param => (typeof param) === "function") : [];
+        const nonDecorators = hasDecorator ? params.filter(param => (typeof param) !== "function") : params;
+        const newProperties = decorators.map((func, index) => {
+          const prop = `_private_${key}_${index}`;
+          Object.defineProperty(target, prop, func(target, prop, desc));
```

Comment:
We probably don't want to modify the prototype like this, since that will be a lasting change. `ember-macro-helpers` already handles a lot of the work for composing computeds, so I think we can probably just 'unwrap' the computeds and pass them directly to the macro helpers version of `@computed`.

Reply:
Correct. `ember-macro-helpers` keeps a proxy object around to define properties on to avoid this situation.